src/pages/SigningOfficialDarRequests.js

```diff
@@ -7,6 +7,7 @@ import {Collections} from '../libs/ajax';
 import { USER_ROLES } from '../libs/utils';
 import { DarCollectionTableColumnOptions, DarCollectionTable } from '../components/dar_collection_table/DarCollectionTable';
 import { consoleTypes } from '../components/dar_collection_table/DarCollectionTableCellData';
+import SigningOfficialDaaAgreementWrapper from '../components/SigningOfficialDaaAgreementWrapper';
```

Comment:
I was very confused by what's happening in the SO console here with the need to wrap the sub each sub tab. I had expected there to be a parent component like we do for all other console pages so we could render individual nav content conditionally. Instead, we have three separate components that all look like a parent console but then behave as a sub nav. I think we need to review this choice as it forces some common logic to be repeated for each tab.

Reply:
I completely agree with you. The issue is the subnav logic is very complex... basically I'm not sure there's a way to say "don't render subnavs, go to this page, then after that render subnavs" - at least without significant refactor of the NavBar code. After talking to Elliot briefly over slack, I thought perhaps it would make sense to remove this entirely from the console pages, and add basically a component wrapper which can be used in the routes. Like so: 

```
    <AuthenticatedRoute path="/signing_official_console/researchers" component={ensureSoHasDaaAcknowledgement(SigningOfficialResearchers)} props={props} rolesAllowed={[USER_ROLES.admin, USER_ROLES.signingOfficial]} />
    <AuthenticatedRoute path="/signing_official_console/dar_requests" component={ensureSoHasDaaAcknowledgement(SigningOfficialDarRequests)} props={props} rolesAllowed={[USER_ROLES.admin, USER_ROLES.signingOfficial]} />
    <AuthenticatedRoute path="/signing_official_console/data_submitters" component={ensureSoHasDaaAcknowledgement(SigningOfficialDataSubmitters)} props={props} rolesAllowed={[USER_ROLES.admin, USER_ROLES.signingOfficial]} />
```

This injects code that will show the DAA acknowledgement page. I also incorporated Elliot's code for storage and caching. Let me know what you think of this approach.